### PR TITLE
store, debian: add golang-gopkg-macaroon.v1 package dependency

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -29,9 +29,19 @@ prepare: |
...
     apt update
     # utilities
-    apt install -y devscripts jq
+    apt install -y devscripts jq  software-properties-common
+
+    # needed so that we have golang-gopkg-macaroon.v1 which is not (yet)
+    # in trusty
+    add-apt-repository --update ppa:snappy-dev/image
+    # this should not be needed but apparently it is :/
+    sudo apt update
```
Comment (pedronis) [inline spread.yaml]: don't need the sudo here

**Comment by pete-woods** [comment]
please retest this

File: store/store.go [hunk 2]
```diff
@@ -170,7 +174,7 @@ func assertsURL() string {
 }
 
 func myappsURL() string {
-	if os.Getenv("SNAPPY_USE_STAGING_MYAPPS") != "" {
+	if os.Getenv("SNAPPY_USE_STAGING_CPI") != "" {
```
Comment (chipaca) [inline store/store.go]: why this? CPI isnt' myapps isn't CPI.

File: overlord/auth/auth.go [hunk 3]
```diff
@@ -161,6 +165,62 @@ func newMacaroonAuthenticator(macaroon string, discharges []string) *MacaroonAut
...
+	if err != nil {
+		return "", err
+	}
+	encoded := base64.URLEncoding.EncodeToString(marshalled)
+	return encoded, nil
+}
+
+// base64Decode decodes base64 data that might be missing trailing pad characters
+// (copied from macaroon package; store serialized macaroons miss trailing padding)
+func base64Decode(b64String string) ([]byte, error) {
```
Comment (chipaca) [inline overlord/auth/auth.go]: why isn't this `base64.RawStdEncoding.DecodeString`?

Comment (matiasb) [inline overlord/auth/auth.go]: Will that handle base64 data missing trailing pad characters? The macaroon library we use server-side is serializing the macaroons data in that situation. This specific helper handles this as expected (while using base64.URLEncoding.Decode was failing).

Comment (chipaca) [inline overlord/auth/auth.go]: From the base64 docs:
> ```
> var RawURLEncoding = URLEncoding.WithPadding(NoPadding)
> ```
> RawURLEncoding is the unpadded alternate base64 encoding defined in RFC 4648. It is typically used in URLs and file names. This is the same as URLEncoding but omits padding characters.

File: store/store.go (hunk 2 above)
Comment (matiasb) [inline store/store.go]: To make it easy to switch to staging server. It doesn't make sense to use staging CPI but prod MyApps (ie. things will not work ok). myappsURL is the base url from which macaroons are got to interact with CPI. I can undo this change, if you think it was better before.

Comment (chipaca) [inline store/store.go]: does the same apply to SAS? Maybe we should move them all under a single variable if so.

File: overlord/auth/auth.go (hunk 3 above)
Comment (matiasb) [inline overlord/auth/auth.go]: Nice, let me confirm that works for our case.

File: store/store.go (hunk 2 above)
Comment (pete-woods) [inline store/store.go]: we should definitely do that at some point

**Comment by pete-woods** [comment]
Matías has merged this PR into his branch.

**Pull request closed without merging**